pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -121,19 +157,13 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 	if err != nil {
 		return err
 	}
-	if platform == PlatformAWS {
-		r.reqLogger.Info("not creating a CephObjectStore because the platform is AWS")
-		return nil
-	}
-	if platform == PlatformGCP {
-		r.reqLogger.Info("not creating a CephObjectStore because the platform is GCP")
-		return nil
-	}
-	if platform == PlatformAzure {
-		r.reqLogger.Info("not creating a CephObjectStore because the platform is Azure")
+	if r.isCloudPlatform(platform) {
+		r.reqLogger.Info("Cloud Platform: '" + string(platform) +
+			"', detected. Not creating CephObjectStore.")
+		// we should exit from this function early itself.
+		instance.Status.CephObjectStoresCreated = true
```

Comment:
@jarrpa , in places where ever we make changes to `instance.Status.<variables>`, are we not supposed to call,
`r.client.Status().Update(context.TODO(), instance)`
method and check the return error status.

Reply:
This is already taken care of in this block, which runs after every `ensure` function: https://github.com/openshift/ocs-operator/blob/master/pkg/controller/storagecluster/reconcile.go#L239-L266